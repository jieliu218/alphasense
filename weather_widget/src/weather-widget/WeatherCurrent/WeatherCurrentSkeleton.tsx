import { Skeleton } from "@mui/material";

export const WeatherCurrentSkeleton = () => {
  return <Skeleton variant="rectangular" width={340} height={153} />;
};

export default WeatherCurrentSkeleton;
