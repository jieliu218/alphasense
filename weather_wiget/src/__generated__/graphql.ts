import { TypedDocumentNode as DocumentNode } from "@graphql-typed-document-node/core";
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = {
  [K in keyof T]: T[K];
};
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & {
  [SubKey in K]?: Maybe<T[SubKey]>;
};
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & {
  [SubKey in K]: Maybe<T[SubKey]>;
};
export type MakeEmpty<
  T extends { [key: string]: unknown },
  K extends keyof T,
> = { [_ in K]?: never };
export type Incremental<T> =
  | T
  | {
      [P in keyof T]?: P extends " $fragmentName" | "__typename" ? T[P] : never;
    };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: { input: string; output: string };
  String: { input: string; output: string };
  Boolean: { input: boolean; output: boolean };
  Int: { input: number; output: number };
  Float: { input: number; output: number };
};

export type CurrentUnits = {
  __typename?: "CurrentUnits";
  interval?: Maybe<Scalars["String"]["output"]>;
  relativeHumidity2m?: Maybe<Scalars["String"]["output"]>;
  temperature2m?: Maybe<Scalars["String"]["output"]>;
  time?: Maybe<Scalars["String"]["output"]>;
  windSpeed10m?: Maybe<Scalars["String"]["output"]>;
};

export type CurrentWeather = {
  __typename?: "CurrentWeather";
  interval?: Maybe<Scalars["Int"]["output"]>;
  relativeHumidity2m?: Maybe<Scalars["Int"]["output"]>;
  temperature2m?: Maybe<Scalars["Float"]["output"]>;
  time?: Maybe<Scalars["String"]["output"]>;
  windSpeed10m?: Maybe<Scalars["Float"]["output"]>;
};

export type HourlyUnits = {
  __typename?: "HourlyUnits";
  relativeHumidity2m?: Maybe<Scalars["String"]["output"]>;
  temperature2m?: Maybe<Scalars["String"]["output"]>;
  time?: Maybe<Scalars["String"]["output"]>;
  windSpeed10m?: Maybe<Scalars["String"]["output"]>;
};

export type HourlyWeather = {
  __typename?: "HourlyWeather";
  relativeHumidity2m?: Maybe<Array<Maybe<Scalars["Int"]["output"]>>>;
  temperature2m?: Maybe<Array<Maybe<Scalars["Float"]["output"]>>>;
  time?: Maybe<Array<Maybe<Scalars["String"]["output"]>>>;
  windSpeed10m?: Maybe<Array<Maybe<Scalars["Float"]["output"]>>>;
};

export type Query = {
  __typename?: "Query";
  getWeatherData?: Maybe<WeatherData>;
};

export type QueryGetWeatherDataArgs = {
  latitude: Scalars["Float"]["input"];
  longitude: Scalars["Float"]["input"];
};

export type WeatherData = {
  __typename?: "WeatherData";
  current?: Maybe<CurrentWeather>;
  currentUnits?: Maybe<CurrentUnits>;
  elevation?: Maybe<Scalars["Float"]["output"]>;
  generationtimeMs?: Maybe<Scalars["Float"]["output"]>;
  hourly?: Maybe<HourlyWeather>;
  hourlyUnits?: Maybe<HourlyUnits>;
  latitude?: Maybe<Scalars["Float"]["output"]>;
  longitude?: Maybe<Scalars["Float"]["output"]>;
  timezone?: Maybe<Scalars["String"]["output"]>;
  timezoneAbbreviation?: Maybe<Scalars["String"]["output"]>;
  utcOffsetSeconds?: Maybe<Scalars["Int"]["output"]>;
};

export type WeatherCurrentFFragment = {
  __typename?: "WeatherData";
  current?: {
    __typename?: "CurrentWeather";
    time?: string | null;
    interval?: number | null;
    temperature2m?: number | null;
    windSpeed10m?: number | null;
    relativeHumidity2m?: number | null;
  } | null;
  currentUnits?: {
    __typename?: "CurrentUnits";
    time?: string | null;
    interval?: string | null;
    temperature2m?: string | null;
    windSpeed10m?: string | null;
    relativeHumidity2m?: string | null;
  } | null;
} & { " $fragmentName"?: "WeatherCurrentFFragment" };

export type WeatherHourlyFFragment = {
  __typename?: "WeatherData";
  hourly?: {
    __typename?: "HourlyWeather";
    time?: Array<string | null> | null;
    temperature2m?: Array<number | null> | null;
    relativeHumidity2m?: Array<number | null> | null;
    windSpeed10m?: Array<number | null> | null;
  } | null;
  hourlyUnits?: {
    __typename?: "HourlyUnits";
    time?: string | null;
    temperature2m?: string | null;
    relativeHumidity2m?: string | null;
    windSpeed10m?: string | null;
  } | null;
} & { " $fragmentName"?: "WeatherHourlyFFragment" };

export type GetWeatherDataQueryVariables = Exact<{
  latitude: Scalars["Float"]["input"];
  longitude: Scalars["Float"]["input"];
  showCurrent: Scalars["Boolean"]["input"];
  showHourly: Scalars["Boolean"]["input"];
}>;

export type GetWeatherDataQuery = {
  __typename?: "Query";
  getWeatherData?:
    | ({
        __typename?: "WeatherData";
        latitude?: number | null;
        longitude?: number | null;
      } & {
        " $fragmentRefs"?: {
          WeatherCurrentFFragment: WeatherCurrentFFragment;
          WeatherHourlyFFragment: WeatherHourlyFFragment;
        };
      })
    | null;
};

export const WeatherCurrentFFragmentDoc = {
  kind: "Document",
  definitions: [
    {
      kind: "FragmentDefinition",
      name: { kind: "Name", value: "WeatherCurrentF" },
      typeCondition: {
        kind: "NamedType",
        name: { kind: "Name", value: "WeatherData" },
      },
      selectionSet: {
        kind: "SelectionSet",
        selections: [
          {
            kind: "Field",
            name: { kind: "Name", value: "current" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                { kind: "Field", name: { kind: "Name", value: "interval" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
              ],
            },
          },
          {
            kind: "Field",
            name: { kind: "Name", value: "currentUnits" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                { kind: "Field", name: { kind: "Name", value: "interval" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
              ],
            },
          },
        ],
      },
    },
  ],
} as unknown as DocumentNode<WeatherCurrentFFragment, unknown>;
export const WeatherHourlyFFragmentDoc = {
  kind: "Document",
  definitions: [
    {
      kind: "FragmentDefinition",
      name: { kind: "Name", value: "WeatherHourlyF" },
      typeCondition: {
        kind: "NamedType",
        name: { kind: "Name", value: "WeatherData" },
      },
      selectionSet: {
        kind: "SelectionSet",
        selections: [
          {
            kind: "Field",
            name: { kind: "Name", value: "hourly" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
              ],
            },
          },
          {
            kind: "Field",
            name: { kind: "Name", value: "hourlyUnits" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
              ],
            },
          },
        ],
      },
    },
  ],
} as unknown as DocumentNode<WeatherHourlyFFragment, unknown>;
export const GetWeatherDataDocument = {
  kind: "Document",
  definitions: [
    {
      kind: "OperationDefinition",
      operation: "query",
      name: { kind: "Name", value: "GetWeatherData" },
      variableDefinitions: [
        {
          kind: "VariableDefinition",
          variable: {
            kind: "Variable",
            name: { kind: "Name", value: "latitude" },
          },
          type: {
            kind: "NonNullType",
            type: { kind: "NamedType", name: { kind: "Name", value: "Float" } },
          },
        },
        {
          kind: "VariableDefinition",
          variable: {
            kind: "Variable",
            name: { kind: "Name", value: "longitude" },
          },
          type: {
            kind: "NonNullType",
            type: { kind: "NamedType", name: { kind: "Name", value: "Float" } },
          },
        },
        {
          kind: "VariableDefinition",
          variable: {
            kind: "Variable",
            name: { kind: "Name", value: "showCurrent" },
          },
          type: {
            kind: "NonNullType",
            type: {
              kind: "NamedType",
              name: { kind: "Name", value: "Boolean" },
            },
          },
        },
        {
          kind: "VariableDefinition",
          variable: {
            kind: "Variable",
            name: { kind: "Name", value: "showHourly" },
          },
          type: {
            kind: "NonNullType",
            type: {
              kind: "NamedType",
              name: { kind: "Name", value: "Boolean" },
            },
          },
        },
      ],
      selectionSet: {
        kind: "SelectionSet",
        selections: [
          {
            kind: "Field",
            name: { kind: "Name", value: "getWeatherData" },
            arguments: [
              {
                kind: "Argument",
                name: { kind: "Name", value: "latitude" },
                value: {
                  kind: "Variable",
                  name: { kind: "Name", value: "latitude" },
                },
              },
              {
                kind: "Argument",
                name: { kind: "Name", value: "longitude" },
                value: {
                  kind: "Variable",
                  name: { kind: "Name", value: "longitude" },
                },
              },
            ],
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "latitude" } },
                { kind: "Field", name: { kind: "Name", value: "longitude" } },
                {
                  kind: "FragmentSpread",
                  name: { kind: "Name", value: "WeatherCurrentF" },
                  directives: [
                    {
                      kind: "Directive",
                      name: { kind: "Name", value: "include" },
                      arguments: [
                        {
                          kind: "Argument",
                          name: { kind: "Name", value: "if" },
                          value: {
                            kind: "Variable",
                            name: { kind: "Name", value: "showCurrent" },
                          },
                        },
                      ],
                    },
                  ],
                },
                {
                  kind: "FragmentSpread",
                  name: { kind: "Name", value: "WeatherHourlyF" },
                  directives: [
                    {
                      kind: "Directive",
                      name: { kind: "Name", value: "include" },
                      arguments: [
                        {
                          kind: "Argument",
                          name: { kind: "Name", value: "if" },
                          value: {
                            kind: "Variable",
                            name: { kind: "Name", value: "showHourly" },
                          },
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          },
        ],
      },
    },
    {
      kind: "FragmentDefinition",
      name: { kind: "Name", value: "WeatherCurrentF" },
      typeCondition: {
        kind: "NamedType",
        name: { kind: "Name", value: "WeatherData" },
      },
      selectionSet: {
        kind: "SelectionSet",
        selections: [
          {
            kind: "Field",
            name: { kind: "Name", value: "current" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                { kind: "Field", name: { kind: "Name", value: "interval" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
              ],
            },
          },
          {
            kind: "Field",
            name: { kind: "Name", value: "currentUnits" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                { kind: "Field", name: { kind: "Name", value: "interval" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
              ],
            },
          },
        ],
      },
    },
    {
      kind: "FragmentDefinition",
      name: { kind: "Name", value: "WeatherHourlyF" },
      typeCondition: {
        kind: "NamedType",
        name: { kind: "Name", value: "WeatherData" },
      },
      selectionSet: {
        kind: "SelectionSet",
        selections: [
          {
            kind: "Field",
            name: { kind: "Name", value: "hourly" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
              ],
            },
          },
          {
            kind: "Field",
            name: { kind: "Name", value: "hourlyUnits" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
              ],
            },
          },
        ],
      },
    },
  ],
} as unknown as DocumentNode<GetWeatherDataQuery, GetWeatherDataQueryVariables>;
