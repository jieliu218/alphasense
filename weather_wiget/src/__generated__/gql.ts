import * as types from "./graphql";
import { TypedDocumentNode as DocumentNode } from "@graphql-typed-document-node/core";

/**
 * Map of all GraphQL operations in the project.
 *
 * This map has several performance disadvantages:
 * 1. It is not tree-shakeable, so it will include all operations in the project.
 * 2. It is not minifiable, so the string of a GraphQL query will be multiple times inside the bundle.
 * 3. It does not support dead code elimination, so it will add unused operations.
 *
 * Therefore it is highly recommended to use the babel or swc plugin for production.
 */
const documents = {
  "\n  fragment WeatherCurrentF on WeatherData {\n    current {\n      time\n      interval\n      temperature2m\n      windSpeed10m\n      relativeHumidity2m\n    }\n    currentUnits {\n      time\n      interval\n      temperature2m\n      windSpeed10m\n      relativeHumidity2m\n    }\n  }\n":
    types.WeatherCurrentFFragmentDoc,
  "\n  fragment WeatherHourlyF on WeatherData {\n    hourly {\n      time\n      temperature2m\n      relativeHumidity2m\n      windSpeed10m\n    }\n    hourlyUnits {\n      time\n      temperature2m\n      relativeHumidity2m\n      windSpeed10m\n    }\n  }\n":
    types.WeatherHourlyFFragmentDoc,
  "\n    query GetWeatherData(\n      $latitude: Float!\n      $longitude: Float!\n      $showCurrent: Boolean!\n      $showHourly: Boolean!\n    ) {\n      getWeatherData(latitude: $latitude, longitude: $longitude) {\n        latitude\n        longitude\n        ...WeatherCurrentF @include(if: $showCurrent)\n        ...WeatherHourlyF @include(if: $showHourly)\n      }\n    }\n  ":
    types.GetWeatherDataDocument,
};

/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 *
 *
 * @example
 * ```ts
 * const query = gql(`query GetUser($id: ID!) { user(id: $id) { name } }`);
 * ```
 *
 * The query argument is unknown!
 * Please regenerate the types.
 */
export function gql(source: string): unknown;

/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function gql(
  source: "\n  fragment WeatherCurrentF on WeatherData {\n    current {\n      time\n      interval\n      temperature2m\n      windSpeed10m\n      relativeHumidity2m\n    }\n    currentUnits {\n      time\n      interval\n      temperature2m\n      windSpeed10m\n      relativeHumidity2m\n    }\n  }\n",
): (typeof documents)["\n  fragment WeatherCurrentF on WeatherData {\n    current {\n      time\n      interval\n      temperature2m\n      windSpeed10m\n      relativeHumidity2m\n    }\n    currentUnits {\n      time\n      interval\n      temperature2m\n      windSpeed10m\n      relativeHumidity2m\n    }\n  }\n"];
/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function gql(
  source: "\n  fragment WeatherHourlyF on WeatherData {\n    hourly {\n      time\n      temperature2m\n      relativeHumidity2m\n      windSpeed10m\n    }\n    hourlyUnits {\n      time\n      temperature2m\n      relativeHumidity2m\n      windSpeed10m\n    }\n  }\n",
): (typeof documents)["\n  fragment WeatherHourlyF on WeatherData {\n    hourly {\n      time\n      temperature2m\n      relativeHumidity2m\n      windSpeed10m\n    }\n    hourlyUnits {\n      time\n      temperature2m\n      relativeHumidity2m\n      windSpeed10m\n    }\n  }\n"];
/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function gql(
  source: "\n    query GetWeatherData(\n      $latitude: Float!\n      $longitude: Float!\n      $showCurrent: Boolean!\n      $showHourly: Boolean!\n    ) {\n      getWeatherData(latitude: $latitude, longitude: $longitude) {\n        latitude\n        longitude\n        ...WeatherCurrentF @include(if: $showCurrent)\n        ...WeatherHourlyF @include(if: $showHourly)\n      }\n    }\n  ",
): (typeof documents)["\n    query GetWeatherData(\n      $latitude: Float!\n      $longitude: Float!\n      $showCurrent: Boolean!\n      $showHourly: Boolean!\n    ) {\n      getWeatherData(latitude: $latitude, longitude: $longitude) {\n        latitude\n        longitude\n        ...WeatherCurrentF @include(if: $showCurrent)\n        ...WeatherHourlyF @include(if: $showHourly)\n      }\n    }\n  "];

export function gql(source: string) {
  return (documents as any)[source] ?? {};
}

export type DocumentType<TDocumentNode extends DocumentNode<any, any>> =
  TDocumentNode extends DocumentNode<infer TType, any> ? TType : never;
