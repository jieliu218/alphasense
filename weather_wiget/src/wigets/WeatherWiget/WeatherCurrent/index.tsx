import {
  Box,
  Divider,
  Skeleton,
  Stack,
  Typography,
  useTheme,
} from "@mui/material";
import ThermostatIcon from "@mui/icons-material/Thermostat";
import AirIcon from "@mui/icons-material/Air";
import OpacityIcon from "@mui/icons-material/Opacity";
import FmdGoodIcon from "@mui/icons-material/FmdGood";
import AccessTimeIcon from "@mui/icons-material/AccessTime";
import { gql, FragmentType, useFragment } from "../../../__generated__";
import { CurrentUnits, CurrentWeather } from "../../../__generated__/graphql";
import { fDateTime } from "../../../utils/formatTime";
import { bgBlur, textBlur } from "../../../theme/css";

export const WeatherCurrentFragment = gql(/* GraphQL */ `
  fragment WeatherCurrentF on WeatherData {
    current {
      time
      interval
      temperature2m
      windSpeed10m
      relativeHumidity2m
    }
    currentUnits {
      time
      interval
      temperature2m
      windSpeed10m
      relativeHumidity2m
    }
  }
`);

const WeatherCurrent = (props: {
  latitude: number;
  longitude: number;
  current: FragmentType<typeof WeatherCurrentFragment>;
  currentUnits: FragmentType<typeof WeatherCurrentFragment>;
}) => {
  const theme = useTheme();

  const current = useFragment(
    WeatherCurrentFragment,
    props.current,
  ) as CurrentWeather;

  const currentUnits = useFragment(
    WeatherCurrentFragment,
    props.currentUnits,
  ) as CurrentUnits;

  const textColor = textBlur(theme.palette.common.white, 0.7);

  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        bgcolor: bgBlur({ color: theme.palette.common.white, opacity: 0.2 }),
        p: 2,
        borderRadius: 2,
      }}
    >
      <Stack
        direction="row"
        justifyContent="space-between"
        alignItems="center"
        spacing={2}
      >
        <Typography
          gutterBottom
          variant="h3"
          component="span"
          color={textColor}
          sx={{ gap: 2 }}
        >
          {`${current?.temperature2m}${currentUnits?.temperature2m}`}
        </Typography>
        <Stack
          direction="column"
          justifyContent="space-between"
          alignItems="flex-start"
          spacing={0.25}
        >
          <Stack
            direction="row"
            justifyContent="space-between"
            alignItems="center"
            spacing={0.25}
          >
            <AccessTimeIcon sx={{ fontSize: 20, color: textColor }} />
            <Typography
              gutterBottom
              variant="body1"
              component="span"
              color={textColor}
            >
              {fDateTime(current?.time, "h:mm aaaa")}
            </Typography>
          </Stack>
          <Stack
            direction="row"
            justifyContent="space-between"
            alignItems="center"
            spacing={0.25}
          >
            <FmdGoodIcon sx={{ fontSize: 20, color: textColor }} />
            <Typography
              gutterBottom
              variant="body1"
              component="span"
              color={textColor}
            >
              {`${props.latitude?.toFixed()}N, ${props.longitude?.toFixed()}E`}
            </Typography>
          </Stack>
        </Stack>
      </Stack>
      <Divider sx={{ mt: 2, mb: 2 }} />
      <Stack
        direction="row"
        justifyContent="space-between"
        alignItems="center"
        spacing={2}
      >
        <Stack
          direction="row"
          justifyContent="space-between"
          alignItems="center"
          spacing={0.25}
        >
          <ThermostatIcon sx={{ fontSize: 20, color: textColor }} />
          <Typography
            gutterBottom
            variant="body1"
            component="span"
            color={textColor}
          >
            {`${current?.temperature2m}${currentUnits?.temperature2m}`}
          </Typography>
        </Stack>
        <Stack
          direction="row"
          justifyContent="space-between"
          alignItems="center"
          spacing={0.25}
        >
          <AirIcon sx={{ fontSize: 20, color: textColor }} />
          <Typography
            gutterBottom
            variant="body1"
            component="span"
            color={textColor}
          >
            {`${current?.windSpeed10m}${currentUnits?.windSpeed10m}`}
          </Typography>
        </Stack>
        <Stack
          direction="row"
          justifyContent="space-between"
          alignItems="center"
          spacing={0.25}
        >
          <OpacityIcon sx={{ fontSize: 20, color: textColor }} />
          <Typography
            gutterBottom
            variant="body1"
            component="span"
            color={textColor}
          >
            {`${current?.relativeHumidity2m}${currentUnits?.relativeHumidity2m}`}
          </Typography>
        </Stack>
      </Stack>
    </Box>
  );
};

export const WeatherCurrentSkeleton = () => {
  return <Skeleton variant="rectangular" width={332} height={145} />;
};

export default WeatherCurrent;
