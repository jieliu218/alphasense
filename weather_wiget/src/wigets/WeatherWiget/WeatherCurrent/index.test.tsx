import { render, screen } from "@testing-library/react";
import WeatherCurrent, { WeatherCurrentFragment } from "./index";
import { FragmentType, useFragment } from "../../../__generated__";

jest.mock("../../../__generated__", () => ({
  useFragment: jest.fn(),
  gql: jest.fn(),
}));

jest.mock("@mui/material", () => ({
  ...jest.requireActual("@mui/material"),
  useTheme: () => ({
    palette: {
      common: {
        white: "#ffffff",
      },
    },
  }),
}));

jest.mock("../../../utils/formatTime", () => ({
  fDateTime: jest.fn().mockReturnValue("Mocked Date"),
}));

const mockCurrent = {
  temperature2m: "20",
  windSpeed10m: "5",
  relativeHumidity2m: "60",
  time: "2022-01-01T12:00:00Z",
} as FragmentType<typeof WeatherCurrentFragment>;

const mockCurrentUnits = {
  temperature2m: "°C",
  windSpeed10m: "km/h",
  relativeHumidity2m: "%",
  time: "ISO8601",
} as FragmentType<typeof WeatherCurrentFragment>;

describe("WeatherCurrent", () => {
  it("renders without crashing and displays weather data", () => {
    (useFragment as jest.Mock)
      .mockReturnValueOnce(mockCurrent)
      .mockReturnValueOnce(mockCurrentUnits);
    const { asFragment } = render(
      <WeatherCurrent
        latitude={45}
        longitude={90}
        current={mockCurrent}
        currentUnits={mockCurrentUnits}
      />,
    );

    expect(screen.getAllByText(/20°C/).length).toBe(2);
    expect(screen.getByText(/Mocked Date/)).toBeInTheDocument();
    expect(screen.getByText(/45N, 90E/)).toBeInTheDocument();
    expect(screen.getByText(/5km\/h/)).toBeInTheDocument();
    expect(screen.getByText(/60%/)).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
