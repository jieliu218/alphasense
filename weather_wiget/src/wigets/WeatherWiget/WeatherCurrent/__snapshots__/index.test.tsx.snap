// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCurrent renders without crashing and displays weather data 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-eksn94"
  >
    <div
      class="MuiStack-root css-1fdhh1r-MuiStack-root"
    >
      <span
        class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1kejgzr-MuiTypography-root"
      >
        20°C
      </span>
      <div
        class="MuiStack-root css-fb11zo-MuiStack-root"
      >
        <div
          class="MuiStack-root css-8og1hd-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-7l7r9c-MuiSvgIcon-root"
            data-testid="AccessTimeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2M12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8"
            />
            <path
              d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-zt1oku-MuiTypography-root"
          >
            Mocked Date
          </span>
        </div>
        <div
          class="MuiStack-root css-8og1hd-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-7l7r9c-MuiSvgIcon-root"
            data-testid="FmdGoodIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2c-4.2 0-8 3.22-8 8.2 0 3.32 2.67 7.25 8 11.8 5.33-4.55 8-8.48 8-11.8C20 5.22 16.2 2 12 2m0 10c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2"
            />
          </svg>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-zt1oku-MuiTypography-root"
          >
            45N, 90E
          </span>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-ctpjxq-MuiDivider-root"
    />
    <div
      class="MuiStack-root css-1fdhh1r-MuiStack-root"
    >
      <div
        class="MuiStack-root css-8og1hd-MuiStack-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-7l7r9c-MuiSvgIcon-root"
          data-testid="ThermostatIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 13V5c0-1.66-1.34-3-3-3S9 3.34 9 5v8c-1.21.91-2 2.37-2 4 0 2.76 2.24 5 5 5s5-2.24 5-5c0-1.63-.79-3.09-2-4m-4-2V5c0-.55.45-1 1-1s1 .45 1 1v1h-1v1h1v2h-1v1h1v1z"
          />
        </svg>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-zt1oku-MuiTypography-root"
        >
          20°C
        </span>
      </div>
      <div
        class="MuiStack-root css-8og1hd-MuiStack-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-7l7r9c-MuiSvgIcon-root"
          data-testid="AirIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14.5 17c0 1.65-1.35 3-3 3s-3-1.35-3-3h2c0 .55.45 1 1 1s1-.45 1-1-.45-1-1-1H2v-2h9.5c1.65 0 3 1.35 3 3M19 6.5C19 4.57 17.43 3 15.5 3S12 4.57 12 6.5h2c0-.83.67-1.5 1.5-1.5s1.5.67 1.5 1.5S16.33 8 15.5 8H2v2h13.5c1.93 0 3.5-1.57 3.5-3.5m-.5 4.5H2v2h16.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5v2c1.93 0 3.5-1.57 3.5-3.5S20.43 11 18.5 11"
          />
        </svg>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-zt1oku-MuiTypography-root"
        >
          5km/h
        </span>
      </div>
      <div
        class="MuiStack-root css-8og1hd-MuiStack-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-7l7r9c-MuiSvgIcon-root"
          data-testid="OpacityIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.66 8 12 2.35 6.34 8C4.78 9.56 4 11.64 4 13.64s.78 4.11 2.34 5.67 3.61 2.35 5.66 2.35 4.1-.79 5.66-2.35S20 15.64 20 13.64 19.22 9.56 17.66 8M6 14c.01-2 .62-3.27 1.76-4.4L12 5.27l4.24 4.38C17.38 10.77 17.99 12 18 14z"
          />
        </svg>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-zt1oku-MuiTypography-root"
        >
          60%
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
